src/symtab/CMakeLists.txt

```diff
@@ -16,6 +16,7 @@ set(SYMTAB_HEADERS
 # =============================================================================
 add_library(symtab_obj OBJECT ${SYMTAB_SOURCES} ${SYMTAB_HEADERS})
 set_property(TARGET symtab_obj PROPERTY POSITION_INDEPENDENT_CODE ON)
+target_link_libraries(symtab_obj PRIVATE fmt::fmt-header-only)
```

Comment:
So, here you are able to set it `PRIVATE` because elsewhere it isn't? Is there maybe a more intuitive way to do this?

Reply:
No, I can set it `PRIVATE` because only `SYMTAB_SOURCES` use fmt, so the include paths and such only need to be set there.  This is the 'clean' way to do things, imo, because it makes the dependencies as localised as possible.  I didn't change the way `nmodl` was getting its `fmt` dependency through `codegen`, because I didn't want to overly re-org stuff.

Reply:
Alright, I see. thanks for the explanation.

Reply:
Turns out this was a bad idea. This is only supported since CMake 3.12, but we support CMake < 3.3 for building nmodl. So that breaks the build for older CMakes...

Reply:
Rough, which part, the `PRIVATE`? They can be dropped.

> but we support CMake < 3.3 for building nmodl

Didn't know that requirement was so low.  Might be worthwhile to set it in the CI, so that this would have failed.

Reply:
Let me correct `<=3.3` and we would need 3.12 for being able to do `target_link_libraries` on OBJECT libraries. We agreed that using by default version 3.10 is OK, but that wouldn't help us here either.

Reply:
and you are right, the CI should have the cmake version that we require to catch this type of issues in the future.

Reply:
> 3.12 for being able to do target_link_libraries on OBJECT libraries

Bah, the uneven distribution of CMake things makes things hard.

It looks like one might be able to work around it w/:
https://cmake.org/cmake/help/v3.10/manual/cmake-buildsystem.7.html#object-libraries

```
Although object libraries may not be named directly in calls to the target_link_libraries() command, they can be “linked” indirectly by using an Interface Library whose INTERFACE_SOURCES target property is set to name $<TARGET_OBJECTS:objlib>.
```
However, I doubt that refactor is worth the time, since (hopefully?) CMake 3.12 will be adopted first?

Reply:
Unfortunately, we can't adopt 3.12, because nmodl is used in coreneuron and neuron and can be built along with them through submodules - and especially for neuron we can't simply bump the cmake version. Thanks for looking up the relevant part in the doc, I'll look into it and create a new PR when ready.

Reply:
> and especially for neuron we can't simply bump the cmake version

Yeah, I figured.  I feel your pain.

> [...] create a new PR when ready.

I personally don't consider this a high priority - I think a general cleanup and possibly looking at the forced `STATIC` would help any CMake refactoring, but that's a tedious job too...so, really, up to you :)